tests/unit/metadata/models/object.js

```diff
@@ -0,0 +1,78 @@
+import assert from 'assert';
+import ObjectMD from '../../../../lib/metadata/models/object/object';
+
+describe('Metadata regular object', () => {
+    let md = null;
+
+    beforeEach(() => {
+        md = new ObjectMD();
+    });
+
+    [
+        ['ModelVersion', null, 2],
+        ['OwnerDisplayName', null, ''],
+        ['OwnerDisplayName', 'owner-display-name'],
+        ['OwnerId', null, ''],
+        ['OwnerId', 'owner-id'],
+        ['CacheControl', null, ''],
+        ['CacheControl', 'cache-control'],
+        ['ContentDisposition', null, ''],
+        ['ContentDisposition', 'content-disposition'],
+        ['ContentEncoding', null, ''],
+        ['ContentEncoding', 'content-encoding'],
+        ['Expires', null, ''],
+        ['Expires', 'expire-date'],
+        ['ContentLength', null, 0],
+        ['ContentLength', 15000],
+        ['ContentType', null, ''],
+        ['ContentType', 'content-type'],
+        ['LastModified', new Date().toJSON()],
+        ['ContentMd5', null, ''],
+        ['ContentMd5', 'content-md5'],
+        ['AmzVersionId', null, 'null'],
+        ['AmzVersionId', 'version-id'],
+        ['AmzServerVersionId', null, ''],
+        ['AmzServerVersionId', 'server-version-id'],
+        ['AmzStorageClass', null, 'STANDARD'],
+        ['AmzStorageClass', 'storage-class'],
+        ['AmzServerSideEncryption', null, ''],
+        ['AmzServerSideEncryption', 'server-side-encryption'],
+        ['AmzEncryptionKeyId', null, ''],
+        ['AmzEncryptionKeyId', 'encryption-key-id'],
+        ['AmzEncryptionCustomerAlgorithm', null, ''],
+        ['AmzEncryptionCustomerAlgorithm', 'customer-algorithm'],
+        ['Acl', null, {
+            Canned: 'private',
+            FULL_CONTROL: [],
+            WRITE_ACP: [],
+            READ: [],
+            READ_ACP: [],
+        }],
+        ['Acl', {
+            Canned: 'public',
+            FULL_CONTROL: ['id'],
+            WRITE_ACP: ['id'],
+            READ: ['id'],
+            READ_ACP: ['id'],
+        }],
+        ['Key', null, ''],
+        ['Key', 'key'],
+        ['Location', null, []],
+        ['Location', ['location1']],
+    ].forEach(test => {
```

Comment:
but why is it you only test default value for many of them? Do you not want to test all the setter methods, or did I misunderstand?